eio/tests/dune

```diff
@@ -10,6 +10,7 @@
     server-ec.pem
     server-ec.key
     (package tls-eio)
+    (package mirage-crypto-rng-eio)
```

Comment:
this change looks like an actual independent bugfix - which is fine to be merged in a separate commit (and PR)

Reply:
I think this is only necessary in a vendoring situation like this PR, thus the change. Outside of the use-case, i.e. vendoring your dependencies - this is not needed as before this PR.